src/plone/restapi/serializer/dxcontent.py

```diff
@@ -41,7 +41,12 @@
 
 @implementer(ISerializeToJson)
 @adapter(IDexterityContent, Interface)
+def get_allow_discussion_value(context, request):
+    return getMultiAdapter((context, request), name="conversation_view").enabled()
```

Comment:
The assignment of `result["allow_discussion"]` must be conditional. see my suggestion at:

https://github.com/plone/plone.restapi/pull/1655#discussion_r1239804188